Update wpt harness

Constellation opened the pull request:
#### 14b85d7e40cb6fb708303553b9408b6274d1d524
<pre>
Update wpt harness
<a href="https://bugs.webkit.org/show_bug.cgi?id=245632">https://bugs.webkit.org/show_bug.cgi?id=245632</a>

Reviewed by Tim Nguyen.

We found that our wpt test harness is too old (one year old) and it does not work with import-maps tests in wpt directory.
This patch updates wpt test harness from revision 4b2dde1f3b77f095012de10f8d60e3937dd7202d.
Some expectations are updated because of testharness / webidl2.js update (some functions do more tests).

* LayoutTests/imported/w3c/web-platform-tests/resources/blank.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/channel.sub.js: Added.
(randInt):
(toHex):
(createUuid):
(SocketCache):
(SocketCache.):
(SocketCache.prototype.async getOrCreate):
(SocketCache.prototype.async close):
(SocketCache.prototype.async closeAll):
(Channel):
(Channel.prototype.hasConnection):
(Channel.prototype.async connect):
(Channel.prototype.async close):
(Channel.prototype.addEventListener):
(Channel.prototype.removeEventListener):
(Channel.prototype._dispatch):
(prototype.async connect):
(prototype.async _send):
(prototype.async send):
(prototype.async disconnectReader):
(prototype.async delete):
(RecvChannel):
(RecvChannel.prototype.async connect):
(RecvChannel.prototype.readMessage):
(RecvChannel.prototype.nextMessage):
(self.channel):
(self.global_channel):
(prototype.async close):
(prototype.async handleMessage):
(prototype.addMessageHandler):
(prototype.removeMessageHandler):
(prototype.nextMessage):
(prototype.setResponseHandler):
(prototype.handleMessage):
(prototype.close):
(RemoteGlobal):
(RemoteGlobal.prototype.async connect):
(RemoteGlobal.prototype.async sendMessage):
(RemoteGlobal.prototype.async call):
(RemoteGlobal.prototype.async postMessage):
(RemoteGlobal.prototype.disconnectReader):
(RemoteGlobal.prototype.close):
(typeName):
(remoteId):
(RemoteObject):
(RemoteObject.from):
(RemoteObject.prototype.toLocal):
(RemoteObject.prototype.delete):
(serialize):
* LayoutTests/imported/w3c/web-platform-tests/resources/check-layout-th.js:
(assert_tolerance):
* LayoutTests/imported/w3c/web-platform-tests/resources/check-layout.js: Added.
(insertAfter):
(positionedAncestor):
(checkSubtreeExpectedValues):
(checkAttribute):
(checkExpectedValues):
(window.checkLayout):
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness-shadowrealm.js:
(promise_setup.):
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness.js:
(IdlArray.prototype.assert_type_is):
(IdlInterface.prototype.test_self):
(IdlInterface.prototype.test_member_iterable):
(IdlInterface.prototype.test_members):
* LayoutTests/imported/w3c/web-platform-tests/resources/sriharness.js:
(SRIScriptTest):
(SRIScriptTest.prototype.execute):
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-actions.js:
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness.js:
(WindowTestEnvironment.prototype._dispatch):
(ShadowRealmTestEnvironment):
(ShadowRealmTestEnvironment.prototype.begin):
(ShadowRealmTestEnvironment.prototype.add_on_loaded_callback):
(create_test_environment):
(EventWatcher):
(promise_setup):
(done): Deleted.
(generate_tests): Deleted.
(step_timeout): Deleted.
(truncate): Deleted.
(is_node): Deleted.
(format_value): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/build.sh: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/README.md:
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/VERSION.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/webidl2.js:
(webpackUniversalModuleDefinition):
(return.):
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/w3c-import.log: Copied from LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/w3c-import.log.

Canonical link: <a href="https://commits.webkit.org/254847@main">https://commits.webkit.org/254847@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/27c47e1de1a441d7608e1147119ef2a8b1e303cf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90447 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/35033 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21089 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/99781 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/157250 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/33532 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/28730 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/82811 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/96225 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96102 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/26662 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77300 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26529 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81455 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81268 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69541 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/34626 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/15314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32449 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16285 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/3391 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36213 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39204 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1465 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38126 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35374 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
lgtm, in the commit message, please mention:
- which WPT upstream revision you imported from
- why there are results outside of the testharness directory being changed (I assume it's due to the test harness updates, but would be nice to know which ones exactly)

Also, please wait for ews results to finish.

Constellation committed:
Update wpt harness
https://bugs.webkit.org/show_bug.cgi?id=245632

Reviewed by Tim Nguyen.

We found that our wpt test harness is too old (one year old) and it does not work with import-maps tests in wpt directory.
This patch updates wpt test harness from revision 4b2dde1f3b77f095012de10f8d60e3937dd7202d.
Some expectations are updated because of testharness / webidl2.js update (some functions do more tests).

* LayoutTests/imported/w3c/web-platform-tests/resources/blank.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/channel.sub.js: Added.
(randInt):
(toHex):
(createUuid):
(SocketCache):
(SocketCache.):
(SocketCache.prototype.async getOrCreate):
(SocketCache.prototype.async close):
(SocketCache.prototype.async closeAll):
(Channel):
(Channel.prototype.hasConnection):
(Channel.prototype.async connect):
(Channel.prototype.async close):
(Channel.prototype.addEventListener):
(Channel.prototype.removeEventListener):
(Channel.prototype._dispatch):
(prototype.async connect):
(prototype.async _send):
(prototype.async send):
(prototype.async disconnectReader):
(prototype.async delete):
(RecvChannel):
(RecvChannel.prototype.async connect):
(RecvChannel.prototype.readMessage):
(RecvChannel.prototype.nextMessage):
(self.channel):
(self.global_channel):
(prototype.async close):
(prototype.async handleMessage):
(prototype.addMessageHandler):
(prototype.removeMessageHandler):
(prototype.nextMessage):
(prototype.setResponseHandler):
(prototype.handleMessage):
(prototype.close):
(RemoteGlobal):
(RemoteGlobal.prototype.async connect):
(RemoteGlobal.prototype.async sendMessage):
(RemoteGlobal.prototype.async call):
(RemoteGlobal.prototype.async postMessage):
(RemoteGlobal.prototype.disconnectReader):
(RemoteGlobal.prototype.close):
(typeName):
(remoteId):
(RemoteObject):
(RemoteObject.from):
(RemoteObject.prototype.toLocal):
(RemoteObject.prototype.delete):
(serialize):
* LayoutTests/imported/w3c/web-platform-tests/resources/check-layout-th.js:
(assert_tolerance):
* LayoutTests/imported/w3c/web-platform-tests/resources/check-layout.js: Added.
(insertAfter):
(positionedAncestor):
(checkSubtreeExpectedValues):
(checkAttribute):
(checkExpectedValues):
(window.checkLayout):
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness-shadowrealm.js:
(promise_setup.):
* LayoutTests/imported/w3c/web-platform-tests/resources/idlharness.js:
(IdlArray.prototype.assert_type_is):
(IdlInterface.prototype.test_self):
(IdlInterface.prototype.test_member_iterable):
(IdlInterface.prototype.test_members):
* LayoutTests/imported/w3c/web-platform-tests/resources/sriharness.js:
(SRIScriptTest):
(SRIScriptTest.prototype.execute):
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-actions.js:
* LayoutTests/imported/w3c/web-platform-tests/resources/testharness.js:
(WindowTestEnvironment.prototype._dispatch):
(ShadowRealmTestEnvironment):
(ShadowRealmTestEnvironment.prototype.begin):
(ShadowRealmTestEnvironment.prototype.add_on_loaded_callback):
(create_test_environment):
(EventWatcher):
(promise_setup):
(done): Deleted.
(generate_tests): Deleted.
(step_timeout): Deleted.
(truncate): Deleted.
(is_node): Deleted.
(format_value): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/build.sh: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/README.md:
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/VERSION.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/webidl2.js:
(webpackUniversalModuleDefinition):
(return.):
* LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/w3c-import.log: Copied from LayoutTests/imported/w3c/web-platform-tests/resources/webidl2/lib/w3c-import.log.

Canonical link: https://commits.webkit.org/254847@main
